File: artifacts/api-server/src/routes/analytics.ts
import { Router } from "express";
import { db, eventsTable, registrationsTable, feedbackTable, usersTable } from "@workspace/db";
import { requireAuth, requireRole } from "../middlewares/auth";

const router = Router();

router.get("/analytics/registrations", requireAuth, requireRole("admin", "faculty"), async (req, res): Promise<void> => {
  const registrations = await db.select().from(registrationsTable);
  res.json(registrations.map(r => ({
    id: r.id,
    eventId: r.eventId,
    studentId: r.studentId,
    paymentStatus: r.paymentStatus,
    adminComment: r.adminComment ?? null,
    registeredAt: r.registeredAt.toISOString(),
  })));
});

router.get("/analytics/feedback", requireAuth, requireRole("admin", "faculty"), async (req, res): Promise<void> => {
  const feedbacks = await db.select().from(feedbackTable);
  res.json(feedbacks.map(f => ({
    id: f.id,
    eventId: f.eventId,
    studentId: f.studentId,
    rating: f.rating,
    comments: f.comments ?? null,
    submittedAt: f.submittedAt.toISOString(),
  })));
});

export default router;

File: artifacts/api-server/routes/analytics.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from database import get_db
import models
from auth_utils import require_roles

router = APIRouter()

@router.get("/analytics/registrations")
def analytics_registrations(
    current_user: models.User = Depends(require_roles("admin", "faculty")),
    db: Session = Depends(get_db),
):
    regs = db.query(models.Registration).all()
    return [
        {
            "id": r.id, "eventId": r.event_id, "studentId": r.student_id,
            "paymentStatus": r.payment_status, "adminComment": r.admin_comment,
            "registeredAt": r.registered_at.isoformat() if r.registered_at else "",
        }
        for r in regs
    ]

@router.get("/analytics/feedback")
def analytics_feedback(
    current_user: models.User = Depends(require_roles("admin", "faculty")),
    db: Session = Depends(get_db),
):
    feedbacks = db.query(models.Feedback).all()
    return [
        {
            "id": f.id, "eventId": f.event_id, "studentId": f.student_id,
            "rating": f.rating, "comments": f.comments,
            "submittedAt": f.submitted_at.isoformat() if f.submitted_at else "",
        }
        for f in feedbacks
    ]
